src/redux/features/apiSlice.js

```diff
@@ -0,0 +1,47 @@
+import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react";
+
+export const apiSlice = createApi({
+  reducerPath: "api",
+  baseQuery: fetchBaseQuery({ baseUrl: "https://dummyjson.com/" }),
+  endpoints: (builder) => ({
+    getProducts: builder.query({
+      query: ({ page = 1, limit = 8 } = {}) =>
+        `products?limit=${limit}&skip=${(page - 1) * limit}`,
+    }),
+
+    getProduct: builder.query({
+      query: (id) => `products/${id}`,
+    }),
+
+    createProduct: builder.mutation({
+      query: (product) => ({
```

Comment:
createProduct: builder.mutation({
  query: (product) => ({
    // Suggestion: The route 'products/add' is not RESTful.
    url: "products/add", // Consider changing this to 'products/' for better adherence to RESTful conventions.
    method: "POST",
    body: product,
  }),
}),